apps/system/js/advanced_telemetry.js

```diff
@@ -209,15 +230,32 @@
     this.collecting = false;
     this.metrics = null;
     this.deviceID = null;
+    // Is the user idle? Updated in handleEvent() based on an idle observer
+    this.idle = false;
   };
 
   AT.prototype.handleEvent = function handleEvent(e) {
     switch (e.type) {
       case HUDMETRICS:
-        if (this.collecting && navigator.onLine) {
-          this.transmit(e.detail);
+        if (this.merge) {
+          this.mergeSend(e.detail, false);
```

Comment:
I'm thinking about the pros and cons of linking the merging of data and the transmitting of data. My first reaction is that this is linking unrelated concepts, that merging and transmitting should be separate. The relevant concepts seem to be: 'is merging required', 'did merging happen', 'are metrics being collected', 'was data transmitted'. The answers will determine whether to persist new metrics (data was merged but not sent) or to clear the persisted data (data was merged and data was transmitted).
I'm thinking the code will be cleaner and more self-documenting if these relevant concepts are expressed separately.